scripts/dts/gen_defines.py

```diff
@@ -354,9 +355,11 @@ def write_special_props(node):
     write_status(node)
 
     if node.parent and "fixed-partitions" in node.parent.compats:
-        macro = f"{node.z_path_id}_PARTITION_ID"
-        out_dt_define(macro, flash_area_num)
-        flash_area_num += 1
+        if node.status == "okay":
```

Comment:
The reason for this change is also not explained in the commit log. It says what is happening, but not why.

What is the goal?

I'm confused because I spot-checked various nodes in the tree and they all look like this:

	partitions {
		compatible = "fixed-partitions";
		boot_partition: partition@0 {
			label = "mcuboot";
			reg = <0x000000000 0x0000C000>;
		};
                /* ... */
	};

Since boot_partition does not have an explicit status, it has status "okay", because that is the behavior when status is missing.

I couldn't find any nodes which have status "disabled" in this PR, so it doesn't seem to have any effect.

What am I missing?

Reply:
It is done so that partitions that get disabled do not get IDs as they will not be included into map.
The macros that include partitions into map add only these that have "okay" status, which means that when storing partitions in array, you may not use ID to index them because the ID is not related status.

Reply:
> It is done so that partitions that get disabled do not get IDs as they will not be included into map.

But no partitions are disabled, so I don't get the point.